spec/services/hubee_api_client_spec.rb

```diff
@@ -0,0 +1,65 @@
+RSpec.describe HubEEAPIClient do
+  let(:hubee_api_client) { described_class.new }
+
+  before do
+    stub_request(:post, hubee_api_client.send(:hubee_auth_url))
+      .to_return(status: 200, headers: { 'Content-Type' => 'application/json' }, body: { "access_token" => "token" }.to_json)
+  end
+
+  describe '#access_token' do
+    subject(:access_token) { hubee_api_client.access_token }
+
+    context 'when the API returns a 200' do
+      before do
+        stub_request(:post, hubee_api_client.send(:hubee_auth_url))
+          .to_return(status: 200, headers: { 'Content-Type' => 'application/json' }, body: { "access_token" => "token" }.to_json)
+      end
+
+      it 'returns the access token' do
+        expect(access_token).to eq("token")
+      end
+    end
+
+    context 'when the API returns a 400' do
+      before do
+        stub_request(:post, hubee_api_client.send(:hubee_auth_url))
+          .to_return(status: 400, headers: { 'Content-Type' => 'application/json' }, body: '')
+      end
+
+      it 'raises an error' do
+        expect { subject }.to raise_error(Faraday::BadRequestError)
+      end
+    end
+  end
+
+  describe '#get_organization' do
+    subject(:get_organization) { hubee_api_client.get_organization(siret, code_commune) }
+
+    let(:siret) { '21920023500014' }
+    let(:code_commune) { '75001' }
+    let(:access_token) { 'token' }
+
+    let(:hubee_organization) { { "branchCode" => "75001", "companyRegister" => "21920023500014", "type" => "SI" } }
+
+    before do
+      stub_request(:get, "#{hubee_api_client.send(:api_host)}/referential/v1/organizations/SI-#{siret}-#{code_commune}")
+        .with(
+          headers: { 'Authorization' => "Bearer #{access_token}" })
+        .to_return(status: 200, body: hubee_organization.to_json, headers: { 'Content-Type' => 'application/json' })
+    end
+
+    context 'when organization already exist' do
```

Comment:
Faut mettre le before dans le context

```suggestion
    context 'when organization already exist' do
      before do
        stub_request(:get, "#{hubee_api_client.send(:api_host)}/referential/v1/organizations/SI-#{siret}-#{code_commune}")
          .with(
            headers: { 'Authorization' => "Bearer #{access_token}" })
          .to_return(status: 200, body: hubee_organization.to_json, headers: { 'Content-Type' => 'application/json' })
      end

```